cglib/src/main/java/net/sf/cglib/core/EmitUtils.java

```diff
@@ -425,7 +425,7 @@ private static void hash_object(CodeEmitter e, Type type, Customizer customizer)
         Label end = e.make_label();
         e.dup();
         e.ifnull(skip);
-        if (customizer != null) {
+        if (customizer != null && !(customizer instanceof FieldTypeCustomizer)) {
```

Comment:
I can do that, however it would require adding additional field for `fieldCustomizer`.

The thing is logic behind old and new customizers is quite different: old customizer augmented hashCode/equals code while new one augments field types and constructor.